src/irmin-pack/unix/gc_worker.ml

```diff
@@ -96,28 +92,18 @@ module Make (Args : Gc_args.S) = struct
         | `Contents key -> (key, false)
         | `Inode key | `Node key -> (key, true)
       in
-      let offset =
-        match Pack_key.to_offset key with
-        | Some offset -> offset
-        | None ->
-            raise
-              (Pack_error (`Node_or_contents_key_is_indexed (string_of_key key)))
-      in
-      schedule offset has_children
-    and schedule offset has_children =
-      Priority_queue.add todos offset has_children
+      schedule key has_children
+    and schedule key has_children =
+      match Node_store.get_offset_exn node_store key with
```

Comment:
`schedule` is also called with commit keys, maybe we should rename the node_store to something more general?